src/session-pool.ts

```diff
@@ -630,8 +635,12 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
    * @returns {Promise<Session>}
    */
   async _acquire(): Promise<Session> {
+    const span = getActiveOrNoopSpan();
     if (!this.isOpen) {
-      throw new GoogleError(errors.Closed);
+      span.addEvent('SessionPool is closed');
+      const err = new GoogleError(errors.Closed);
+      setSpanErrorAndException(span, err);
```

Comment:
As per our discussion you are planning to create `tracewrapper` which will handle all the errors and event. 

Will this error be covered in that?

Reply:
Indeed, it'll handle setting the error. I suppose let me undo this setSpanErrorAndException here.